src/apis/NetInfo/index.js

```diff
@@ -63,7 +64,9 @@ const NetInfo = {
       };
     }
 
-    connection.addEventListener(eventTypesMap[type], handler);
+    const wrappedHandler = e => handler(e.target);
```

Comment:
An object of the wrong type is passed to the handler. `type: ConnectionType` is missing, and `onchange` should not be included. A type check or test would be helpful to catch this.